routers/api/v1/api.go

```diff
@@ -1596,6 +1585,47 @@ func Routes() *web.Route {
 	return m
 }
 
+// actionAPI is a struct that holds the actions API
+type actionAPI struct {
+	ListActionsSecrets   func(ctx *context.APIContext)
+	CreateOrUpdateSecret func(ctx *context.APIContext)
+	DeleteSecret         func(ctx *context.APIContext)
+	ListVariables        func(ctx *context.APIContext)
+	GetVariable          func(ctx *context.APIContext)
+	DeleteVariable       func(ctx *context.APIContext)
+	CreateVariable       func(ctx *context.APIContext)
+	UpdateVariable       func(ctx *context.APIContext)
+	GetRegistrationToken func(ctx *context.APIContext)
+}
```

Comment:
This struct looks quite strange.

I think there are some samples for such usage, eg: 

https://github.com/go-gitea/gitea/blob/dd2aaadce3ecd3134a1ba0c82c5aaa05d6c11b2b/routers/web/web.go#L442-L448

Reply:
I will try it.

Reply:
@wxiaoguang lint error: lines are duplicate

Maybe we can change the action API struct to interface?

Reply:
No idea why "lines are duplicate" would happen. Such code should indeed avoid duplication.

Reply:
![image](https://github.com/go-gitea/gitea/assets/21979/bb73ffad-b5b2-4b9c-a77f-43b88f95e912)

![image](https://github.com/go-gitea/gitea/assets/21979/1f0dd7d0-d77c-42ef-949e-b9619a56d4a0)

@wxiaoguang  See.

Reply:
Since don't duplicate them, only use one common function?